### PR TITLE
91: MVP Keycloak UI

### PR DESCRIPTION
**Pull request by collinpreston**
# What are the relevant tickets?
https://github.com/mitodl/mit-open/issues/91

# Description (What does it do?)
Updates the ol Keycloak theme to align with the mockups here: https://www.figma.com/proto/LPnWeOp4ALDqhENz15bZZV/Single-Sign-On---SSO?type=design&node-id=204-650&t=0nvLvdJSyB7WQnrx-0&scaling=scale-down&page-id=0%3A1

# Screenshots (if appropriate):
https://www.figma.com/proto/LPnWeOp4ALDqhENz15bZZV/Single-Sign-On---SSO?type=design&node-id=204-650&t=0nvLvdJSyB7WQnrx-0&scaling=scale-down&page-id=0%3A1

# How can this be tested?
After building and copying the theme and extensions jar files, using the instructions described in the readme of this repository, login to the Keycloak Admin Console.  

1. Update the realm login theme to `ol`.
2. Duplicate the browser flow.  Update the duplicate flow to be:
Screenshot 2023-12-01 at 11 12 39 AM
3. Bind the duplicate browser flow.
4. Create a dummy SAML identity provider (does not need to link to an actual identity provider).
5. Create a second user who is linked to the identity provider and has a verified email address.
6. Setup Keycloak with an SMTP email service.
7. Log out of the Keycloak Admin Console.
8. Test the process flows documented here: https://docs.google.com/document/d/17tJ-C2EwWoSpJWZKjuhMVgsqGtyPH0IN9KakXvSKU0M/edit?pli=1#heading=h.ratoog6xmca6
9. Verify that UI is similar to the mockups described here https://www.figma.com/proto/LPnWeOp4ALDqhENz15bZZV/Single-Sign-On---SSO?type=design&node-id=204-650&t=0nvLvdJSyB7WQnrx-0&scaling=scale-down&page-id=0%3A1
10. Verify that the screens display well on mobile displays.

# Additional Context

1. Some animations and styling are absent due to the complexity to implement them within Keycloak.

### CODE REVIEW

**Comment by collinpreston** [comment]
https://github.com/mitodl/ol-infrastructure/pull/1950 must be merged prior to merging and releasing this PR.

File: ol-keycloak/oltheme/src/main/resources/theme/ol/login/login-password.ftl [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        <div id="kc-form">
+            <div id="kc-form-wrapper">
+                <form id="kc-form-login" onsubmit="login.disabled = true; return true;" action="${url.loginAction}"
+                      method="post">
+                    <div class="${properties.kcFormGroupClass!} no-bottom-margin">
+                        <label for="password" class="${properties.kcLabelClass!}">${msg("password")}</label>
+                        <div class="${properties.kcInputGroup!}">
+                            <input tabindex="2" id="password" class="${properties.kcInputClass!}" name="password"
+                                   type="password" autocomplete="on" autofocus
+                                   aria-invalid="<#if messagesPerField.existsError('password')>true</#if>"
```
Comment (ChristopherChudzicki) [inline ol-keycloak/oltheme/src/main/resources/theme/ol/login/login-password.ftl:15]: This looks like it's either `aria-invalid=""` or `aria-invalid="true"` depending on validity?  

**Suggestion:** Would it be possible to omit the attribute entirely if the field is not yet marked as invalid? 

I don't know if an empty string is problematic, but it's not one of the [4 valid values](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-invalid#values).

File: ol-keycloak/oltheme/src/main/resources/theme/ol/login/register.ftl [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        <#if realm.password && (social.providers?? || unlinkedProviders??)>
+            <div class="textline">or</div>
+        </#if>
+        <#if realm.password && social.providers??>
+            <div id="kc-social-providers" class="${properties.kcFormSocialAccountSectionClass!}">
+                <ul class="${properties.kcFormSocialAccountListClass!} <#if social.providers?size gt 3>${properties.kcFormSocialAccountListGridClass!}</#if>">
+                    <#list social.providers as p>
+                        <li>
+                            <a id="social-${p.alias}" class="${properties.kcFormSocialAccountListButtonClass!} <#if social.providers?size gt 3>${properties.kcFormSocialAccountGridItem!}</#if>"
+                                    type="button" href="${p.loginUrl}">
```
[the remark this answers is not in the document]
Comment (collinpreston) [reply ol-keycloak/oltheme/src/main/resources/theme/ol/login/register.ftl:167]: This is OOTB code.  If it proves to be an issue during functional testing I will update it.

File: ol-keycloak/oltheme/src/main/resources/theme/ol/login/login-password.ftl (hunk 1 above)
Comment (collinpreston) [reply ol-keycloak/oltheme/src/main/resources/theme/ol/login/login-password.ftl:15]: This is OOTB code.  If it proves to be an issue during functional testing I will update it.

**Pull request merged**